wallet/src/de/schildbach/wallet/ui/UsernameSearchResultsAdapter.kt

```diff
@@ -62,22 +62,28 @@ class UsernameSearchResultsAdapter() : RecyclerView.Adapter<UsernameSearchResult
         fun bind(usernameSearchResult: UsernameSearchResult) {
             val defaultAvatar = UserAvatarPlaceholderDrawable.getDrawable(itemView.context,
                     usernameSearchResult.username[0])
-            username.text = usernameSearchResult.username
-            if (usernameSearchResult.profileDocument != null) {
-                usernameSearchResult.profileDocument.apply {
-                    displayName.text = data["displayName"] as String
-                    Glide.with(avatar).load(data["avatarUrl"]).circleCrop()
-                            .placeholder(defaultAvatar).into(avatar)
-                }
+
+            val dashPayProfile = usernameSearchResult.dashPayProfile
+            if (dashPayProfile.displayName.isEmpty()) {
+                displayName.text = dashPayProfile.username
+                username.text = ""
+            } else {
+                displayName.text = dashPayProfile.displayName
+                username.text = usernameSearchResult.username
+            }
+
+            if(dashPayProfile.avatarUrl.isNotEmpty()) {
+                Glide.with(avatar).load(dashPayProfile.avatarUrl).circleCrop()
+                        .placeholder(defaultAvatar).into(avatar)
```

Comment:
In this section, I am copying the scheme from the from the contact and contact request rows.  The first line is the Display Name and the second line is the username, but if there is not display name, then the username is displayed on top and the second line is blank.